Src/WindowsServer/WindowsServer.Shared/MetricManager.cs

```diff
@@ -0,0 +1,1571 @@
+        internal double Min { get; private set; }
+
+        /// <summary>
+        /// Gets maximum sample value.
+        /// </summary>
+        internal double Max { get; private set; }
+
+        /// <summary>
+        /// Gets arithmetic average value in the population.
+        /// </summary>
+        internal double Average
+        {
+            get
+            {
+                return this.Count == 0 ? 0 : this.Sum / this.Count;
+            }
+        }
+
+        /// <summary>
+        /// Gets variance of the values in the population.
+        /// </summary>
+        internal double Variance
+        {
+            get
+            {
+                return this.Count == 0 ? 0 : (this.SumOfSquares / this.Count) - (this.Average * this.Average);
+            }
+        }
+
+        /// <summary>
+        /// Gets standard deviation of the values in the population.
+        /// </summary>
+        internal double StandardDeviation
+        {
+            get
+            {
+                return Math.Sqrt(this.Variance);
+            }
+        }
+
+        /// <summary>
+        /// Adds a value to the time series.
+        /// </summary>
+        /// <param name="value">Metric value.</param>
+        public void Track(double value)
+        {
+            bool lockAcquired = false;
+
+            try
+            {
+                this.trackLock.Enter(ref lockAcquired);
+
+                if ((this.Count == 0) || (value < this.Min))
+                {
+                    this.Min = value;
+                }
+
+                if ((this.Count == 0) || (value > this.Max))
+                {
+                    this.Max = value;
+                }
+
+                this.Count++;
+                this.Sum += value;
+                this.SumOfSquares += value * value;
+            }
+            finally
+            {
+                if (lockAcquired)
+                {
+                    this.trackLock.Exit();
+                }
+            }
+        }
+    }
+
+    internal static class MetricTerms
+    {
+        private const string MetricPropertiesNamePrefix = "_MS";
+
+        public static class Aggregation
+        {
+            public static class Interval
+            {
+                public static class Moniker
+                {
+                    public const string Key = MetricPropertiesNamePrefix + ".AggregationIntervalMs";
+                }
+            }
+        }
+
+        public static class Extraction
+        {
+            public static class ProcessedByExtractors
+            {
+                public static class Moniker
+                {
+                    public const string Key = MetricPropertiesNamePrefix + ".ProcessedByMetricExtractors";
+                    public const string ExtractorInfoTemplate = "(Name:'{0}', Ver:'{1}')";      // $"(Name:'{ExtractorName}', Ver:'{ExtractorVersion}')"
+                }
+            }
+        }
+
+        public static class Autocollection
```

Comment:
I didn't go through each class that I copied to determine if parts of it could be removed. The Autocollection class is part of the MetricTerms class that I copied. So I'd rather not start down this path of removing what was there. Worst case, the code is unused.